app/models/server/models/Users.js

```diff
@@ -447,7 +447,7 @@ export class Users extends Base {
 			username = new RegExp(`^${ s.escapeRegExp(username) }$`, 'i');
 		}
 
-		const query = { username, [`services.${ serviceName }.id`]: serviceName };
+		const query = { username, [`services.${ serviceName }.id`]: username };
```

Comment:
the actual value that is set to `services.${ serviceName }.id` is `serviceData.id` and not the `username` as you can see here: https://github.com/RocketChat/Rocket.Chat/blob/develop/app/custom-oauth/server/custom_oauth_server.js#L391

or am I missing something?

Reply:
the code never gets beyond here because it always returns '!user'.  https://github.com/RocketChat/Rocket.Chat/blob/develop/app/custom-oauth/server/custom_oauth_server.js#L370

I instrumented and tested the fix to verify what was happening.